### PR TITLE
add grpc secure option

### LINKED ISSUE #5551: [package] grpc/1.37.1: TLS doesn't work because it links against grpc++_unsecure instead of grpc++
### Package and Environment Details
  * Package Name/Version: **grpc/1.37.1**
  * Operating System+version: **Windows 10**
  * Compiler+version: **Visual Studio 16**
  * Conan version: **conan 1.35.2**

### Details
When this package is used, it makes the application to link against unsecure and secure versions of Grpc libraries simultaneously, which breaks TLS functionality. When you try to use TLS you get the following error in the console:

```
D0518 10:47:09.802000000 30164 init_unsecure.cc:30] Using insecure gRPC build. Security handshakers will not be invoked even if secure credentials are used.
```

### Steps to reproduce
Add this conan package as a dependency and look at the generated `conanbuildinfo.cmake` file's `CONAN_LIBS_GRPC` variable, it will contain secure and unsecure versions of gRPC libs.

### Cmake logs
[cmake generate cache output.txt](https://github.com/conan-io/conan-center-index/files/6499795/cmake.generate.cache.output.txt)

### Workaround
Look at an [answer from @drmacdon](https://github.com/conan-io/conan-center-index/issues/5551#issuecomment-981263965)

### PR DESCRIPTION
**Pull request by yuriy-mochurad**
Specify library name and version:  **grpc/1.39.1**

Fixes #5551 

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by yuriy-mochurad** [comment]
@ericLemanissier I am not sure this is the correct way to address this issue, but it is fixing it on my end. Wdyt?

**Comment by ericLemanissier** [comment]
I don't have a strong opinion, but this seems like an artificial constraint, because the package binary still contains the same libraries. The true solution is to choose the targets you link to as suggested in https://github.com/conan-io/conan-center-index/issues/5551#issuecomment-894832965

**Review by SSE4 — APPROVED** [review]

File: recipes/grpc/all/conanfile.py [hunk 1]
```diff
@@ -27,7 +27,8 @@ class grpcConan(ConanFile):
         "objective_c_plugin": [True, False],
         "php_plugin": [True, False],
         "python_plugin": [True, False],
-        "ruby_plugin": [True, False]
+        "ruby_plugin": [True, False],
+        "secure": [True, False]
```
Comment (madebr) [inline recipes/grpc/all/conanfile.py:31]: If I read this recipe correct, the build system unconditionally builds both unsecure and secure libraries.
Does it have a flag to enable/disable one of both?

If there isn't, you can do two things:
- When having `secure=True`, remove the unsecure libraries in `package` and vice versa for `secure=False`
- Or, as you're doing now, add a unsecure target when choosing `secure=False`.
  You need to remove the creation of the secure targets in this case.
  Also, it seems like the package for `secure=True` and `secure=False` is the same, so you can do:
    ````python
    def package_id(self):
        del self.info.options.secure
    ````

Comment (SSE4) [reply recipes/grpc/all/conanfile.py:31]: @madebr I think it's almost the same case as we have in google test (and other testing/mocking libraries), where it produces `gtest` and `gtest_main` always, but you are free to link with any of them (depends if you have your own main or just use provided). AFAIK SDK2 and few other libraries use similar approaches.

**Comment by SSE4** [comment]
please rebase, there are conflicts

**Comment by yuriy-mochurad** [comment]
@lkotsonis Please, review this PR

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]
https://github.com/grpc/grpc/blob/2d4f3c56001cd1e1f85734b2f7c5ce5f2797c38a/tools/buildgen/plugins/verify_duplicate_sources.py#L27

**Comment by lkotsonis** [comment]
> @lkotsonis Please, review this PR

Looks good, don't have permission to review I think :)

**Comment by yuriy-mochurad** [comment]
@uilianries @ericLemanissier could you, please review this PR? I would like to merge it into master. Thank you!

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/grpc/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/grpc/all/conanfile.py:29]: I'll take this review into account in https://github.com/conan-io/conan-center-index/pull/9044:
- remove secure in package_id()

**Pull request merged**